### PR TITLE
feat: respect time range when building parquet reader

### PR DESCRIPTION
**Pull request by v0y4g3r**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
- #3944

## What's changed and what's your intention?

This PR converts the desired time range into `SimpleFilterContext` while building parquet readers so that the reader can filter out those rows outside the desired time range.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/mito2/src/read/scan_region.rs [hunk 1]
```diff
@@ -269,6 +269,7 @@ impl ScanRegion {
         );
 
         let index_applier = self.build_index_applier();
+        info!("Request filters: {:?}", self.request.filters);
```
Comment (evenyag) [inline src/mito2/src/read/scan_region.rs:272]: Do we need this log? Maybe we can change it to `debug!()`.

File: src/mito2/src/sst/parquet/reader.rs [hunk 2]
```diff
@@ -225,7 +229,7 @@ impl ParquetReaderBuilder {
 
         metrics.build_cost = start.elapsed();
 
-        let filters = if let Some(predicate) = &self.predicate {
+        let mut filters = if let Some(predicate) = &self.predicate {
```
Comment (evenyag) [inline src/mito2/src/sst/parquet/reader.rs:232]: Is it possible to reserve the capacity for `filters`?

File: src/query/src/tests/time_range_filter_test.rs [hunk 3]
```diff
@@ -114,9 +114,9 @@ struct TimeRangeTester {
 impl TimeRangeTester {
     async fn check(&self, sql: &str, expect: TimestampRange) {
         let _ = exec_selection(self.engine.clone(), sql).await;
-        let filters = self.get_filters();
+        let mut filters = self.get_filters();
```
Comment (evenyag) [inline src/query/src/tests/time_range_filter_test.rs:117]: Should it be `take_filters()`?

File: src/mito2/src/sst/parquet/reader.rs (hunk 2 above)
Comment (v0y4g3r) [reply src/mito2/src/sst/parquet/reader.rs:232]: > Is it possible to reserve the capacity for `filters`?

Cannot determine the num of time range filters here.

File: src/table/src/predicate.rs [hunk 4]
```diff
@@ -131,250 +131,249 @@ impl Predicate {
...
 }
 
-// tests for `TimeRangePredicateBuilder` locates in src/query/tests/time_range_filter_test.rs
+// tests for `build_time_range_predicate` locates in src/query/tests/time_range_filter_test.rs
 // since it requires query engine to convert sql to filters.
-/// `TimeRangePredicateBuilder` extracts time range from logical exprs to facilitate fast
+/// `build_time_range_predicate` extracts time range from logical exprs to facilitate fast
 /// time range pruning.
-pub struct TimeRangePredicateBuilder<'a> {
+pub fn build_time_range_predicate<'a>(
```
[the remark this answers is not in the document]
Comment (v0y4g3r) [reply src/table/src/predicate.rs:138]: > Could we port more tests for this function? I remember that we had many tests for it before.

Most cases are already here: https://github.com/GreptimeTeam/greptimedb/commit/43aefc5d74dfa73b7819cae77b7eb546d8534a41

**Review by waynexia — APPROVED** [review]

**Review by evenyag — APPROVED** [review]

**Pull request merged**